src/libponyc/pass/names.c

```diff
@@ -178,6 +178,15 @@ static bool names_typeparam(pass_opt_t* opt, ast_t** astp, ast_t* def)
     return false;
   }
 
+  if (ast_id(ast) == TK_NOMINAL) {
+    ast_t* module = ast_nearest(ast, TK_MODULE);
+    if (module && ast_get(module, ast_name(id), 0)) {
+      ast_error(opt->check.errors, def,
+                "Type parameter shadows existing type");
```

Comment:
The code style for this error message should be just indented two spaces in from the previous line, as we do in the `ast_error` in the code block above.